Riot/Modules/TabBar/MasterTabBarController.m

```diff
@@ -336,6 +336,12 @@ - (void)selectTabAtIndex:(MasterTabBarIndex)tabBarIndex
 {
     NSInteger index = [self indexOfTabItemWithTag:tabBarIndex];
     self.selectedIndex = index;
+    
+    // Normally the title gets updated in the UITabBarControllerDelegate.
+    // But that gets called only when the user selects a tab. When a tab get
+    // set otherwise (e.g. after changing language) we have to update it here.
+    [self.selectedViewController viewWillAppear:NO];
+    titleView.titleLabel.text = self.selectedViewController.accessibilityLabel;
```

Comment:
Hi, thank you for raising this. 
I have a small question though: wouldn't the correct solution here be to use the viewController's title property and stop them from updating their parent tabbar navigation item titles directly?

Reply:
I agree, the current way of updating the title seems not intuitive. But i am not familiar enough with these elements to change the whole system. Currently the title property of the tab views is not set and updated at all. And i don't know how these titles would then be shown in the navigation bar.
That is why my code changes use the current way of updating the title, with the least risk of breaking other functionality.

Reply:
Okay, fair enough. I'm going to leave this with @langleyd.